src/CTA.FeatureDetection.ProjectType/CompiledFeatures/VBClassLibraryFeature.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Codelyzer.Analysis;
+using CTA.FeatureDetection.Common.Extensions;
+using CTA.FeatureDetection.Common.Models.Features.Base;
+using CTA.Rules.Config;
+using Microsoft.Build.Construction;
+
+namespace CTA.FeatureDetection.ProjectType.CompiledFeatures
+{
+    public class VBClassLibraryFeature : CompiledFeature
+    {
+        /// <summary>
+        /// Determines if a project is an VB class library project by looking at file extensions and nuget references
+        /// </summary>
+        /// <param name="analyzerResult"></param>
+        /// <returns>Whether a project is an VB class library project or not</returns>
+        public override bool IsPresent(AnalyzerResult analyzerResult)
+        {
+            var project = analyzerResult.ProjectResult;
+            var isVBProj = project.ContainsFileWithExtension(Constants.VbClassExtension, true) && project.ProjectFilePath.EndsWith(Constants.VbProjExtension)
```

Comment:
should we split out "isVBProj" and the feature specifically? especially if the .vbproj stuff is reused in the others?

Reply:
sure I can make that change in the subsequent PR